packages/wonder-blocks-cell/src/util/types.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+import * as React from "react";
+
+import type {StyleType} from "@khanacademy/wonder-blocks-core";
+import {type Typography} from "@khanacademy/wonder-blocks-typography";
+
+/**
+ * A set of values that can be used to configure the horizontal rule appearance.
+ */
+type HorizontalRuleVariant = "full-width" | "inset" | "none";
+
+/**
+ * A subset of CSS Properties to allow overriding some of the default styles set
+ * on the accessory wrapper (loosely based on StyleType).
+ */
+export type AccessoryStyle = {|
+    minWidth?: 16 | 24 | 32 | 48,
+    alignSelf?: "flex-start" | "flex-end" | "center",
+    justifyContent?: "flex-start" | "flex-end" | "center",
+|};
+
+/**
+ * A union that allows using  plain text or WB Typography elements.
+ */
+type TypographyText = string | React.Element<Typography>;
+
+/**
+ * Common properties for all cells.
+ */
+type SharedProps = {|
+    /**
+     * The title / main content of the cell. You can either provide a string or
+     * a Typography component.
+     */
+    title: TypographyText,
+
+    /**
+     * Adds a horizontal rule at the bottom of the cell that can be used to
+     * separate cells within groups such as lists. Defaults to `inset`.
+     */
+    horizontalRule?: HorizontalRuleVariant,
+
+    /**
+     * Optional custom styles applied to the cell container.
+     */
+    style?: StyleType,
+
+    /**
+     * Optional test ID for e2e testing.
+     */
+    testId?: string,
+|};
+
+/**
+ * We define these props as a union to enforce that `leftAccessoryStyle` and/or
+ * `rightAccessoryStyle` are passed in only if `leftAccessory` and/or
+ * `rightAccessory` are set.
+ */
+export type CellProps =
+    /**
+     * leftAccessory
+     */
+    | {|
+          ...SharedProps,
+          leftAccessory?: React.Node,
+      |}
```

Comment:
Do we need to do this kind of `CellProps` where it's a union of exact types? Feels like we will have _way_ too many combos once we add in DetailCell/subtitle, etc.

Reply:
Yep, sadly it's one of the downsides of using this approach, we need to cover all these cases. I'm going to try it out this week with the DetailCell to see how it goes.

We could also use a single type (making leftAccessoryStyle? optional directly in `SharedProps`), but then we would got into the "make the impossible states unrepresentable" opposite case, because there would be the case where devs don't set `leftAccessory`, but set a value for `leftAccessoryStyle`.